certbot/certbot/tests/util.py

```diff
@@ -149,43 +148,71 @@ def make_lineage(config_dir, testfile, ec=False):
     return conf_path
 
 
-def patch_get_utility(target='zope.component.getUtility'):
-    """Patch zope.component.getUtility to use a special mock IDisplay.
+def patch_display_service():
+    """Patch CertbotService display property to use a special mock IDisplay.
 
     The mock IDisplay works like a regular mock object, except it also
     also asserts that methods are called with valid arguments.
 
-    :param str target: path to patch
-
-    :returns: mock zope.component.getUtility
+    :returns: mock certbot.display.util._get_display
     :rtype: mock.MagicMock
 
     """
-    return mock.patch(target, new_callable=_create_get_utility_mock)
+    return mock.patch('certbot.display.util._get_display',
+                      new_callable=_create_display_service_mock)
 
 
-def patch_get_utility_with_stdout(target='zope.component.getUtility',
-                                  stdout=None):
-    """Patch zope.component.getUtility to use a special mock IDisplay.
+def patch_display_service_with_stdout(stdout=None):
+    """Patch CertbotService display property to use a special mock IDisplay.
 
     The mock IDisplay works like a regular mock object, except it also
     also asserts that methods are called with valid arguments.
 
     The `message` argument passed to the IDisplay methods is passed to
     stdout's write method.
 
-    :param str target: path to patch
     :param object stdout: object to write standard output to; it is
         expected to have a `write` method
 
-    :returns: mock zope.component.getUtility
+    :returns: mock certbot.display.util._get_display
     :rtype: mock.MagicMock
 
     """
     stdout = stdout if stdout else io.StringIO()
 
-    freezable_mock = _create_get_utility_mock_with_stdout(stdout)
-    return mock.patch(target, new=freezable_mock)
+    return mock.patch('certbot.display.util._get_display',
+                      new=_create_display_service_mock_with_stdout(stdout))
+
+
+def patch_get_utility(target='zope.component.getUtility'):  # pylint: disable=unused-argument
+    """Deprecated, use patch_display_service instead.
+
+    :param str target: path to patch (warning, value is ignored due to deprecation)
+
+    :returns: mock certbot.display.util._get_display
+    :rtype: mock.MagicMock
+
+    """
+    warnings.warn('Decorator certbot.tests.util.patch_display_service is deprecated, '
+                  'use certbot.tests.util.patch_display_service instead.')
+    return patch_display_service()
+
+
+def patch_get_utility_with_stdout(target='zope.component.getUtility',  # pylint: disable=unused-argument
+                                  stdout=None):
```

Comment:
I think these functions should continue to patch `zope.component.getUtility` and leave `_get_display` untouched.

Wherever we choose to mock here, I think there's the potential for things to break, but I think the approach of continuing to patch `zope.component.getUtility` will only break if the user of the function was assuming our code would call `getUtility` internally which isn't an assumption I think they should make. If their own code was calling `getUtility`, leaving these functions as is (with the warning you added) will continue to work until they've updated their code to use the new interface.

Reply:
Really what I'd like to do here is patch both functions, but I don't see a good way to do that.

Reply:
Regarding the discussion below, shouldn't we consider that the refactoring planned in #8948 overrides the possible changes to do on that matter in this PR?

Reply:
For the reason I described in my first post in [this thread](https://github.com/certbot/certbot/pull/8835/files#r671754133), I personally would still like to leave these functions essentially unmodified except for the warning you added.

Reply:
I reverted to the previous behavior.